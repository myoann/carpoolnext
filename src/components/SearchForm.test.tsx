import React from "react";
import { render, screen } from "@testing-library/react";

import SearchForm from "./SearchForm";

jest.mock("../utils", () => ({
  isMobileDevice: () => false,
}));

jest.mock("react-datepicker", () => {
  return function DatePicker() {
    return <input type="text" value="01/01/2024" onChange={jest.fn()} />;
  };
});

describe("SearchForm", () => {
  it("matches the snapshot", () => {
    const { container } = render(<SearchForm />);

    expect(container).toMatchSnapshot();
  });

  it("renders a form", () => {
    render(<SearchForm />);

    expect(
      screen.getByRole("form", { name: "Search for a carpool" }),
    ).toBeInTheDocument();
  });

  it("renders three inputs", () => {
    render(<SearchForm />);

    const placesAutocompleteInputs = screen.getAllByRole("textbox");
    expect(placesAutocompleteInputs).toHaveLength(3);
  });

  it("renders a button with an img inside", () => {
    render(<SearchForm />);

    const button = screen.getByRole("button");
    expect(button).toBeInTheDocument();

    const img = screen.getByRole("img", { name: "Search" });
    expect(img).toBeInTheDocument();

    expect(button).toContainElement(img);
  });
});
