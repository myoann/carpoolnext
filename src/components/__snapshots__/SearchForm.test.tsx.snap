// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm matches the snapshot 1`] = `
<div>
  <form
    aria-label="Search for a carpool"
    autocomplete="off"
    class="searchForm"
  >
    <label
      for="fromCityInput"
    >
      From city
      <div
        class="placesAutocomplete"
      >
        <input
          id="fromCityInput"
          placeholder="Enter a departure place"
          value=""
        />
        <ul />
      </div>
    </label>
    <label
      for="toCityInput"
    >
      To city
      <div
        class="placesAutocomplete"
      >
        <input
          id="toCityInput"
          placeholder="Enter a destination place"
          value=""
        />
        <ul />
      </div>
    </label>
    <label
      for="dateInput"
    >
      Date
      <input
        type="text"
        value="01/01/2024"
      />
    </label>
    <button
      class="submitButton"
      type="submit"
    >
      <img
        alt="Search"
        class="searchIcon"
        data-nimg="1"
        decoding="async"
        height="24"
        loading="lazy"
        src="/search.svg"
        style="color: transparent;"
        width="24"
      />
    </button>
  </form>
</div>
`;
